src/test/kotlin/com/expedia/graphql/execution/FunctionDataFetcherTest.kt

```diff
@@ -126,4 +136,34 @@ internal class FunctionDataFetcherTest {
         assertTrue(result is CompletableFuture<*>)
         assertEquals(expected = "hello", actual = result.get())
     }
+
+    @Test
+    fun `throwException function propagates the original exception`() {
+        val dataFetcher = FunctionDataFetcher(target = MyClass(), fn = MyClass::throwException)
+        val mockEnvironmet: DataFetchingEnvironment = mockk()
+
+        try {
+            dataFetcher.get(mockEnvironmet)
+            assertFalse(true, "Should not be here")
+        } catch (e: Exception) {
+            assertEquals(e.message, "Test Exception")
+        }
+    }
+
+    @Test
+    fun `suspendThrow throws exception when resolved`() {
+        val dataFetcher = FunctionDataFetcher(target = MyClass(), fn = MyClass::suspendThrow)
+        val mockEnvironmet: DataFetchingEnvironment = mockk()
+
+        try {
+            val result = dataFetcher.get(mockEnvironmet)
+            assertTrue(result is CompletableFuture<*>)
```

Comment:
while test still works it only works so because it throws exception in line above, can we remove line 160 and 161 as they are leftovers from `CompletableFuture` return type?

Reply:
It doesn't look like it. When debugging I see the type of `result` as `CompletableFuture` still. Without checking the type, the compiler won't let me call `get`. Removing 161 causes the assertion on line 162 to fail.

Is there a more idiomatic way this should be done?

Reply:
Ah indeed! We are wrapping coroutine results in completable future so we can return it async (so your return type in this case is `CompletableFuture<String>`). Totally forgot about it.